src/components/organisms/NavBarSchedule/NavBarSchedule.tsx

```diff
@@ -77,15 +79,29 @@ export const NavBarSchedule: React.FC<NavBarScheduleProps> = ({
   });
 
   const scheduledStartDate = sovereignVenue?.start_utc_seconds;
+  const scheduledEndDate = sovereignVenue?.end_utc_seconds;
 
-  const firstDayOfSchedule = useMemo(() => {
-    return scheduledStartDate
+  const { firstDayOfSchedule, isScheduleTimeshifted } = useMemo(() => {
+    const today = startOfToday();
+    const firstDayOfEvent = scheduledStartDate
       ? startOfDay(fromUnixTime(scheduledStartDate))
-      : startOfToday();
-  }, [scheduledStartDate]);
+      : today;
+
+    const isScheduleTimeshifted = isBefore(today, firstDayOfEvent);
 
-  const isScheduleTimeshifted = !isToday(firstDayOfSchedule);
+    const firstDayOfSchedule = isScheduleTimeshifted ? firstDayOfEvent : today;
 
+    return {
+      firstDayOfSchedule,
+      isScheduleTimeshifted,
+    };
+  }, [scheduledStartDate]);
```

Comment:
[note] Previously `firstDayOfSchedule` would always be set to the date of `scheduledStartDate` (aka `start_utc_seconds`) when configured. This meant that the schedule would never advance forward each day, and would be able to show at most `SCHEDULE_SHOW_DAYS_AHEAD` days before being unable to show 'today' anymore.

The logic used for `isScheduleTimeshifted` also seemed strange/likely buggy.

Reply:
What happens when  `today` is greater than the last day of the event? Any UI/UX concerns about accessing historical data?

Reply:
@sparkletown/product and/or @sunny-viktoryia is probably best to weigh in here, but my understanding is that the feature as originally implemented/designed would ‘move forward’ and not show past days. I believe this got broken with some open source contributed work.